File: sese/line_to_json_.js
const { once } = require("node:events");
const fs = require("node:fs");
const readline = require("node:readline");

async function main() {
  const args = process.argv.slice(1);
  if (args.length != 3) {
    console.error(`args.length != 3 which args.length = ${args.length}`);
    console.log(`Usage: ${args[0]} prefix path`);
    process.exit(1);
  }

  const prefix = args[1];
  const path = args[2];

  const fileStream = fs.createReadStream(path);

  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity,
  });

  const key = `${prefix}_原始值`;

  rl.on("line", (line) => {
    if (line.length != 0) {
      console.log(JSON.stringify({ key: line }));
    }
  });

  await once(rl, "close");
}

main();

File: sese/line_to_json_.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <filesystem>

#include <nlohmann/json.hpp>

int main(int argc, char **argv)
{
    if (argc != 3)
    {
        std::cerr << "argc != 3 which argc = " << argc << std::endl;
        std::cout << "Usage: " << argv[0] << " prefix file_path" << std::endl;
        exit(1);
    }

    const auto prefix = std::filesystem::path(argv[1]).u8string();
    const std::string file_path = argv[2];

    const auto key_str = prefix + u8"_原始值";
    std::string_view key((const char *)key_str.data(), key_str.length());

    std::ifstream fs(file_path, std::ios::in | std::ios::binary);
    std::string line;
    bool exit = false;
    while (!exit)
    {
        std::getline(fs, line);
        if (!fs)
            exit = true;

        if (!line.empty())
        {
            std::cout
                << nlohmann::json({{key, line}}).dump() << "\n";
        }
    }

    fs.close();
}
